### PR TITLE
Validating the certificate thumbprint

### PR DESCRIPTION
**Pull request by ppathan**
# Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] The code follows the code style and conventions of this project.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

# Description of the change

Validate the thumbprint of the ssl certificate to allow/disallow the connection

### CODE REVIEW

File: ProxyAgent/HttpClient/HttpClient.cs [hunk 1]
```diff
@@ -220,11 +225,24 @@ public HttpClient(ILogger logger)
...
-        private static void SetServerSslSecurity(IHttpRequest request, HttpClientHandler clientHandler)
+        private void SetServerSslSecurity(IHttpRequest request, HttpClientHandler clientHandler)
         {
             if (request.Options.AllowInsecureSslServer && request.UsesSsl())
             {
-                clientHandler.ServerCertificateCustomValidationCallback = delegate { return true; };
+                clientHandler.ServerCertificateCustomValidationCallback += delegate(HttpRequestMessage sender, X509Certificate2 cert, X509Chain chain, SslPolicyErrors error)
+                {
+                    var sslThumbprint = cert.Thumbprint.ToLowerInvariant();
+                    var configThumbprint = this.config.Thumbprint.ToLowerInvariant();
```
Comment (dluc) [inline ProxyAgent/HttpClient/HttpClient.cs]: nit: trim the string, that hit me several times when copying thumbprints from Windows certificate store

File: ProxyAgent/HttpClient/HttpClient.cs [hunk 2]
```diff
@@ -220,11 +225,24 @@ public HttpClient(ILogger logger)
...
+                    if (sslThumbprint  != configThumbprint)
+                    {
+                        this.log.Error("The thumbprint provided by the client don't match", () => new { sslThumbprint, configThumbprint });
```
Comment (dluc) [inline ProxyAgent/HttpClient/HttpClient.cs]: nit: change the msg to "The remote endpoint is using an unknown/invalid SSL certificate, the thumbprint of the certificate doesn't match the value in the configuration"

File: ProxyAgent/appsettings.ini [hunk 3]
```diff
@@ -2,6 +2,7 @@
 ; The remote endpoint where requests are forwarded to
 ; The REMOTE_ENDPOINT env var is set on PCS deployments
 endpoint = "${REMOTE_ENDPOINT}"
+thumbprint = "${REMOTE_ENDPOINT_SSL_THUMBPRINT}"
```
Comment (dluc) [inline ProxyAgent/appsettings.ini]: nit: rename to "ssl_cert_thumbprint"

Comment (ppathan) [reply ProxyAgent/appsettings.ini]: Won't rename this since it requires touching a bunch of files for deployment.

**Pull request merged**